### ISSUE TITLE
[Subscription] filter subscriptions by field doesn't work

### ISSUE THREAD

**Issue by bob-lee** [aws-amplify/amplify-cli#3047]
**Describe the bug**
I have 3 types - `Company`, `Job`, `Jobline` related as below. I am successfully getting real time updates on `Job` filtered by parent company id. So if I am on company A, am getting updates on jobs only for company A not for company B. Nice.
```
+---------+  1:many  +--------------+  1:many  +---------------------+
|COMAPNY  |----------|JOB           |----------|JOBLINE              |
| id      |          | id           |          | id                  |
|         |          | jobCompanyId |          | joblineJobCompanyId |
+---------+          +--------------+          +---------------------+
```
My problem is about subscription for `Jobline` by grand parent company id. Basically I am getting `onUpdateJobline` for ALL companies.

**Amplify CLI Version**
3.17.0

**To Reproduce**
My schema looks like:
```
type Company
  @model(subscriptions: null)
{
  id: ID!
  companyName: String!
  jobs: [Job] @connection(name: "CompanyJobs")
}
type Job
  @model(subscriptions: null)
{
  id: ID!
  orderRef: String
  jobCompanyId: ID!
  company: Company @connection(name: "CompanyJobs")
}
type Jobline
  @model(subscriptions: null)
 {
  id: ID!
  quantity: Int
  joblineJobId: ID!
  job: Job @connection(name: "JobJoblines")
  joblineJobCompanyId: String # connected company id
}

type Subscription {
	onUpdateJob(jobCompanyId: ID): Job
		@aws_subscribe(mutations: ["updateJob"])
	onUpdateJobline(joblineJobCompanyId: String): Jobline
		@aws_subscribe(mutations: ["updateJobline"])
}
```
When I am creating jobline, I am manually recording grand parent company id as `joblineJobCompanyId`.

My update mutations look like:
```
  updateJob(input: {
    id: "69f6099f-f12f-4288-a2d4-fc89a0195e69"
    orderRef: "129"
  }) {
    id 
    orderRef
    jobCompanyId
  }

  updateJobline(input: {
    id: "4a5ac5ac-53d3-4a41-9d99-c055d9caab2b"
    quantity: 25
  }) {
    id
    quantity
    joblineJobCompanyId
  }
```
**Expected behavior**
getting `onUpdateJobline` only for subscribed company id

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Node v10.16.2

**Comment by SwaySway** [comment]
Hello @bob-lee

Are you running into this issue still? Using the following schema I added an extra `@connection` to job for joblines and I was only able to get subscription results based on the joblineJobCompanyId when setting up the subscription

```gql
type Company
  @model(subscriptions: null)
{
  id: ID!
  companyName: String!
  jobs: [Job] @connection(name: "CompanyJobs")
}
type Job
  @model(subscriptions: null)
{
  id: ID!
  orderRef: String
  jobCompanyId: ID!
  company: Company @connection(name: "CompanyJobs")
  joblines: [Jobline] @connection(name: "JobJoblines")
}
type Jobline
  @model(subscriptions: null)
{
  id: ID!
  quantity: Int
  joblineJobId: ID!
  job: Job @connection(name: "JobJoblines")
  joblineJobCompanyId: String # connected company id
}

type Subscription {
	onUpdateJob(jobCompanyId: ID): Job
		@aws_subscribe(mutations: ["updateJob"])
	onUpdateJobline(joblineJobCompanyId: String): Jobline
		@aws_subscribe(mutations: ["updateJobline"])
}
```

**Comment by bob-lee** [comment]
@SwaySway you spotted well that I missed Job.joblines in my original post (too simplified). Anyway I retested and found the issue has gone. I mean I am getting real time updates on joblines only for subscribed company as expected. Originally jobline subscription was anoymous then changed to be by company. I guess that change left some unwanted dangling stuff on the cloud refusing to be by company? CLI version hasn't changed, only clue I can think of is that recently schema has been changed a lot including renaming Job and Jobline types to something else and all those changes clear things up and make it work as expected. Thanks I am closing this issue.
